data/block/metaBlockV3.go

```diff
@@ -262,21 +262,43 @@ func (m *MetaBlockV3) GetMiniBlockHeadersWithDst(destID uint32) map[string]uint3
 			continue
 		}
 
-		for _, val := range m.ShardInfo[i].ShardMiniBlockHeaders {
-			if val.ReceiverShardID == destID && val.SenderShardID != destID {
-				hashDst[string(val.Hash)] = val.SenderShardID
-			}
-		}
+		addShardMBHeadersMBToDestMap(m.ShardInfo[i].ShardMiniBlockHeaders, hashDst, destID)
+	}
+
+	for _, execResults := range m.ExecutionResults {
+		addMetaMBHeadersMBToDestMap(execResults.MiniBlockHeaders, hashDst, destID)
 	}
 
-	for _, val := range m.MiniBlockHeaders {
-		isDestinationShard := (val.ReceiverShardID == destID ||
-			val.ReceiverShardID == core.AllShardId) &&
-			val.SenderShardID != destID
+	return hashDst
+}
+
+func addMetaMBHeadersMBToDestMap(miniBlockHeaders []MiniBlockHeader, hashDst map[string]uint32, destID uint32) {
+	for _, mbHeader := range miniBlockHeaders {
+		isDestinationShard := (mbHeader.ReceiverShardID == destID ||
+			mbHeader.ReceiverShardID == core.AllShardId) &&
+			mbHeader.SenderShardID != destID
 		if isDestinationShard {
-			hashDst[string(val.Hash)] = val.SenderShardID
+			hashDst[string(mbHeader.Hash)] = mbHeader.SenderShardID
 		}
 	}
+}
+
+// GetProposedMiniBlockHeadersWithDst returns a map of hashes and sender IDs for proposed mini blocks
+func (m *MetaBlockV3) GetProposedMiniBlockHeadersWithDst(destID uint32) map[string]uint32 {
+	if m == nil {
+		return nil
+	}
+
+	hashDst := make(map[string]uint32)
+	for i := 0; i < len(m.ShardInfo); i++ {
+		if m.ShardInfo[i].ShardID == destID {
+			continue
+		}
+
+		addShardMBHeadersMBToDestMap(m.ShardInfo[i].ShardMiniBlockHeaders, hashDst, destID)
```

Comment:
in ShardInfo on MetablockV3 we only have the already executed shard miniblocks/shard headers so I don't think these should be added here as well

In ShardInfoProposal we have the headers that were notarized, but there we don't have the miniblocks as well collected, so I would say to just skip the shard miniblock headers.